File: src/main/java/com/nickcoblentz/montoya/settings/BooleanExtensionSetting.java
package com.nickcoblentz.montoya.settings;

import burp.api.montoya.MontoyaApi;

public class BooleanExtensionSetting extends GenericExtensionSetting<Boolean> {

    public BooleanExtensionSetting(MontoyaApi api, String name, String storageKey, Boolean defaultValue, ExtensionSettingSaveLocation defaultSaveLocation) {
        super(api, name, storageKey, defaultValue, defaultSaveLocation);
    }

    @Override
    public void loadFromProject() {
        setCurrentValue(getApi().persistence().extensionData().getBoolean(getStorageKey()));
    }

    @Override
    public void loadFromPreferences() {
        setCurrentValue(getApi().persistence().preferences().getBoolean(getStorageKey()));
    }

    @Override
    public Boolean parseString(String value) {
        return Boolean.valueOf(value);
    }

    public void setCurrentValue(String value) {
        if(value!=null)
            setCurrentValue(parseString(value));
    }
}

File: src/main/java/com/nickcoblentz/montoya/settings/BooleanExtensionSetting.kt
package com.nickcoblentz.montoya.settings

import burp.api.montoya.MontoyaApi

class BooleanExtensionSetting(
    api: MontoyaApi,
    name: String,
    storageKey: String,
    defaultValue: Boolean,
    defaultSaveLocation: ExtensionSettingSaveLocation
) :
    GenericExtensionSetting<Boolean>(api, name, storageKey, defaultValue, defaultSaveLocation) {
    override fun loadFromProject() {
        currentValue = api.persistence().extensionData().getBoolean(storageKey)
    }

    override fun loadFromPreferences() {
        currentValue = api.persistence().preferences().getBoolean(storageKey)
    }

    override fun saveToProject() {
        api.persistence().extensionData().setBoolean(storageKey, currentValue)
    }

    override fun saveToPreferences() {
        api.persistence().preferences().setBoolean(storageKey, currentValue)
    }

    override fun parseString(value: String): Boolean {
        return value.toBoolean()
    }

    override fun setCurrentValue(value: String?) {
        if (value != null) currentValue = parseString(value)
    }
}